pilot/pkg/model/authentication.go

```diff
@@ -33,13 +35,22 @@ import (
 func GetConsolidateAuthenticationPolicy(mesh *meshconfig.MeshConfig, store IstioConfigStore, hostname string, port *Port) *authn.Policy {
 	config := store.AuthenticationPolicyByDestination(hostname, port)
 	if config == nil {
```

Comment:
nit. This may be simpler
```go
...
  if config != nil {
    policy := config.Spec.(*authn.Policy)
    if err := jwksresolver.SetAuthenticationPolicyJwksURIs(policy); err != nil {
      return policy
    }
  }

  return constructLegacyAuthenticationPolicy(mesh, hostname, port)
}

```